src/sql/workbench/electron-browser/modelComponents/diffeditor.component.ts

```diff
@@ -79,8 +81,20 @@ export default class DiffEditorComponent extends ComponentBase implements ICompo
 		this.editorUriRight = uri2.toString();
 
 		let cancellationTokenSource = new CancellationTokenSource();
-		let editorinput1 = this._instantiationService.createInstance(UntitledEditorInput, uri1, false, 'plaintext', '', '');
-		let editorinput2 = this._instantiationService.createInstance(UntitledEditorInput, uri2, false, 'plaintext', '', '');
+		this._textModelService.registerTextModelContentProvider('diffEditor', {
+			provideTextContent: (resource: URI): Promise<ITextModel> => {
+				if (resource.scheme === 'diffEditor') {
+					let modelContent = '';
```

Comment:
Is this check necessary? Given that you're registering this method to the diffEditor scheme already I would think this would only ever be called for URIs with that scheme

Reply:
removed check. I just had this since that's how it was in the test I copied this from, but it works fine without it.